diff --git a/internal/config/tls/tls.go b/internal/config/tls/tls.go
--- a/internal/config/tls/tls.go
+++ b/internal/config/tls/tls.go
@@ -137,7 +137,7 @@ func (t *TLS) GetConfig() (*tls.Config, error) {
 		return nil, errors.Wrap(err, "failed to parse certificate")
 	}
 
-	if time.Now().Add(31 * 24 * time.Hour).After(c.Leaf.NotAfter) {
+	if time.Now().Add(24 * time.Hour).After(c.Leaf.NotAfter) {
 		return nil, ErrTLSCertExpiresSoon
 	}
 
